azurerm/resource_arm_mssql_elasticpool.go

```diff
@@ -177,81 +186,188 @@ func resourceArmMsSqlElasticPool() *schema.Resource {
 		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
 
 			name, _ := diff.GetOk("sku.0.name")
+			tier, _ := diff.GetOk("sku.0.tier")
 			capacity, _ := diff.GetOk("sku.0.capacity")
+			family, _ := diff.GetOk("sku.0.family")
+			maxSizeBytes, _ := diff.GetOk("max_size_bytes")
+			maxSizeGb, _ := diff.GetOk("max_size_gb")
 			minCapacity, _ := diff.GetOk("per_database_settings.0.min_capacity")
 			maxCapacity, _ := diff.GetOk("per_database_settings.0.max_capacity")
 
-			if strings.HasPrefix(strings.ToLower(name.(string)), "gp_") {
+			// Convert Bytes to Gigabytes
+			if maxSizeBytes != 0 {
+				maxSizeGb = float64(maxSizeBytes.(int) / 1073741824)
+			}
+
+			// Basic Checks
+			if strings.EqualFold(name.(string), "BasicPool") {
```

Comment:
Given that this *must* be a specific value maybe it would be best to just not allow the user to set it, or mark it as computed and populate it ourselves with a default?

Reply:
Talked about this in slack, going to leave this as is for... reasons.